internal/testutil/acc/factory.go

```diff
@@ -61,6 +61,16 @@ func ConnV2UsingGov() *admin.APIClient {
 	return client.(*config.MongoDBClient).AtlasV2
 }
 
+func ConnV2UsingResourcePolicy() *admin.APIClient {
+	cfg := config.Config{
+		PublicKey:  os.Getenv("MONGODB_ATLAS_RP_PUBLIC_KEY"),
```

Comment:
knit alternatively you could change the  GH action to be like this, so it get the credentials in the normal env vars: 
```
resource_policy:
...
      - name: Acceptance Tests
        env:
        MONGODB_ATLAS_PUBLIC_KEY: ${{  ${{ secrets.mongodb_atlas_rp_public_key }} }}
         MONGODB_ATLAS_PRIVATE_KEY: ${{  ${{ secrets.mongodb_atlas_rp_private_key }} }}
          MONGODB_ATLAS_LAST_VERSION: ${{ needs.get-provider-version.outputs.provider_version }}
          MONGODB_ATLAS_ENABLE_PREVIEW: "true"
          ACCTEST_PACKAGES: |
            ./internal/service/resourcepolicy
        run: make testacc

```

Reply:
env vars like MONGODB_ATLAS_PUBLIC_KEY defined in the job will override env vars defined for all the workflow

Reply:
Great idea! It avoids any extra provider config or env-vars. You only need to be aware of not running more tests in that job. 
Changed in 77b1c43c3243c8a3e98d4cff79a6af19d7744139